### ISSUE TITLE
Non-Babel project adds babel files in new project

### ISSUE THREAD

**Issue by slanden** [vuejs/vue-cli#4575]
### Version
3.11.0

### Environment info
```
System:
    OS: Windows 10
    CPU: (8) x64 Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
  Binaries:
    Node: 10.16.3 - C:\Program Files\nodejs\node.EXE
    Yarn: Not Found
    npm: 6.11.3 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: 44.18362.329.0
  npmGlobalPackages:
    @vue/cli: Not Found
```

### Steps to reproduce
1. `vue create test-app`
2. select manual presets
3. select presets: PWA, Router, Vuex, Linter, Unit Testing
4. History mode: Yes
5. select: ESLint   Prettier
6. select: Lint On Save
7. select: Jest
8. select: In dedicated config files

### What is expected?
The resulting new project should contain only what's used by the selected presets and mandatory Vue files, nothing extra.

Upon viewing the new project's files, I know what everything is and everything is there because I specifically asked Vue to put it there.

### What is actually happening?
Extra files which seem to be files used by packages or libraries not installed, appear in the new project's directory.

Although I'm mostly confident I don't need them, I'm not 100% certain of hidden consequences for deleting them.

i.e.

- what is .babelrc? babel bla bla, don't need it - delete.
- what is .browserlistrc? don't know, don't need it - delete.
- what is postcss.config.js? don't know, don't need it - delete.
- why does my package.json have `babel-eslint` devDependency? don't know, wasted space, don't need it - `npm un babel-eslint`

---
This is not that important relative to the rest of the work the Vue team is doing, but it does mean every new project (according to the reproduction steps above) needs to have these files deleted and the `babel-eslint` package uninstalled if the goal is to keep the project minimal.

**Comment by LinusBorg** [comment]
> what is .babelrc? 

Are you sure? We would generate babel.config.js maybe, but not babelrc?

> what is postcss.config.js?

We always add postcss for applying autoprefixer. It's documented and it's not an option you can un-select right now. If you think it should be, open a featue request.

> what is .browserlistrc?

autoprefixer needs this file to know what your targets browsers are.

> why does my package.json have babel-eslint devDependency? 

That can indeed be fixed in the eslint plugin.

**Comment by slanden** [comment]
> Are you sure? We would generate babel.config.js maybe, but not babelrc?
Positive. Below is the contents of the generated babelrc:
`{
  "plugins": [
    "transform-es2015-modules-commonjs"
  ]
}`

> We always add postcss for applying autoprefixer.
Since Vue CLI uses this internally, it's fine. Being that it is a default that cannot be turned off, it would be nice if that it was tucked away from my project files, but again it's not a big deal.

If the babel-eslint package was fixed in v4 that's enough for me, I'll just keep deleting it until I can upgrade to v4.

**Comment by sodatea** [comment]
> Positive. Below is the contents of the generated babelrc:
> { "plugins": [ "transform-es2015-modules-commonjs" ] }

Working on that. Maybe we can hide the details in a jest preset. For now, it's required for jest & babel-jest to properly process es module syntax.

> Since Vue CLI uses this internally, it's fine. Being that it is a default that cannot be turned off, it would be nice if it was tucked away from my project files, but again it's not a big deal.

When scaffolding the project, the last prompt is "Where do you prefer placing config for Babel, PostCSS, ESLint, etc.?" If "In package.json" is chosen, browserslist & postcss config will be placed in package.json.

> `Parsing error: Unexpected token import`

The dynamic import syntax wasn't supported in ESLint until very recently with the release of espree v6.1.0 last month. Before that, `babel-eslint` is necessary for any project that uses `import()`, so we added it by default in case users are confused by this issue.
We still need time to fully update to ESLint 6 for all our config options. And contributions are welcome.

**Comment by slanden** [comment]
@sodatea thanks for the clarification on those requirements, it seems like all those "extra" files are in fact not extra after all... yet.

> If "In package.json" is chosen, browserslist & postcss config will be placed in package.json.

True, but I should have specified that by "tucked away from my project files" I meant to not exist at all on the surface, if it's a default and must exist somewhere it'd be nice if it was on a lower level like in a Vue CLI preset or config somewhere in its node_modules folder. I'm not sure what the pros/cons of that would be though.

**Comment by sodatea** [comment]
This issue has been fixed as of v4.2 😄
`babel-jest` is hidden behind a Jest preset, and `babel-eslint` is no longer mandatory with the new `ecmaVersion: 2020` parserOption of ESLint.
